xbmc/pvr/PVRDatabase.cpp

```diff
@@ -488,36 +490,18 @@ int CPVRDatabase::Get(CPVRChannelGroup &group)
 
 bool CPVRDatabase::PersistChannels(CPVRChannelGroup &group)
 {
-  bool bReturn(true);
-
-  for (PVR_CHANNEL_GROUP_MEMBERS::iterator it = group.m_members.begin(); it != group.m_members.end(); ++it)
+  for (auto &member : group.m_members)
   {
-    if (it->second.channel->IsChanged() || it->second.channel->IsNew())
-    {
-      if (Persist(*it->second.channel))
-      {
-        it->second.channel->Persisted();
-        bReturn = true;
-      }
-    }
-  }
-
-  bReturn &= CommitInsertQueries();
+    const CPVRChannelPtr channel = member.second.channel;
```

Comment:
Don't get me wrong, I don't want to bother you. I just look things up that I don't understand at the first glance. Here I did not understand the const qualifier. It means that this code won't change the shared_ptr itself but it can change the object it points to, CPVRChannel. I don't think this is what you had in mind, right?